### PR TITLE
Added certora CI workflow [PE-11]

### PR DESCRIPTION
**Pull request by iamsahu**
## Description

This PR integrates the certora verification into the CI. The prover would run on any new changes being pushed to the repo.

[Here](https://github.com/liquid-collective/liquid-collective-protocol/pull/303) is the original PR created by the Certora team. We have just recreated the original PR.

## Notice

- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/liquid-collective/liquid-collective-protocol/pulls) for the same update/change?
- [x] Have you assigned this PR to yourself?
- [x] Have you added at least 1 reviewer?
- [ ] Have you updated the official documentation?
- [ ] Have you added sufficient documentation in your code?
- [ ] Have you added relevant tests to the official test suite?

## Pull Request Type

- [ ] 💫 New Feature (Breaking Change)
- [ ] 💫 New Feature (Non-breaking Change)
- [ ] 🛠️ Bug fix (Non-breaking Change: Fixes an issue)
- [x] 🕹️ Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)

## Testing

- [x] Have you tested this code with the official test suite?
- [ ] Have you tested this code manually?

### CODE REVIEW

File: certora/specs/RiverV1.spec [hunk 1]
```diff
@@ -0,0 +1,408 @@
...
+    depositAndTransfer(e2, recipient) at initial;
+    mathint shares2 = balanceOf(recipient);
+
+    depositAndTransfer(eSum, recipient) at initial;
+    mathint sharesSum = balanceOf(recipient);
+
+    assert shares1 + shares2 <= sharesSum + sharesBefore;
+}
+
+// TODO:
```
Comment (juliaaschmidt) [inline certora/specs/RiverV1.spec:327]: more todos here

File: certora/specs/RiverV1.spec [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+    mathint amount2;
+    address recipient;
+
+    requireInvariant noAssetsNoShares();
+    requireInvariant noAssetsNoSharesUser(recipient);
+
+    require amount1 == to_mathint(e1.msg.value);
+    require amount2 == to_mathint(e2.msg.value);
+    require amount1 + amount2 == to_mathint(eSum.msg.value);
+    // require amount1 == 500;
```
Comment (juliaaschmidt) [inline certora/specs/RiverV1.spec:348]: remove

File: certora/specs/SharesManagerV1.spec [hunk 3]
```diff
@@ -0,0 +1,371 @@
...
+
+function ghostUpdate_onDeposits(uint256 amount) returns bool
+{
+    total_onDeposits = total_onDeposits + amount;
+    return true;
+}
+
+// Partially proved here (with some sanity fails and eror for claimRedeemRequests):
+// https://vaas-stg.certora.com/output/40577/e0f17bcec1594338ad6a53b432cdb04a/?anonymousKey=28d9005d1fe0729067a0749a6fdc28dd043ced24
+// TODO: filter oracle change and remove the ' + total_onEarnings'
```
Comment (juliaaschmidt) [inline certora/specs/SharesManagerV1.spec:50]: remove this unfinished proof?

File: certora/specs/SharesManagerV1.spec [hunk 4]
```diff
@@ -0,0 +1,371 @@
...
+ // + (consensus layer + execution layer rewards)
+ // - (fees + penalties)) * ConversionRate
+ // I.e. given a conversion rate and balances, that LsEth balance/total supply is correct The same invariant for every owner's balance
+
+// TODO: Add _pullCLFunds (consensus layer rewards) tracking
+// TODO: Add _onEarnings (consensus layer rewards) tracking - takes rewards as an argument and computes fees from it
+
+// @title The allowance can only be changed by functions listed in the filter:
+// initRiverV1_1, setConsensusLayerData, decreaseAllowance, increaseAllowance, approve, transferFrom
+// Almost fixed. Latest run:
```
Comment (juliaaschmidt) [inline certora/specs/SharesManagerV1.spec:95]: lol

**Review by juliaaschmidt — COMMENTED** [review]
Not sure we're actually changing any of it? 

Did they write a proof for the mitigated front-running attack? Didn't see it, but the PR is massive.

**Review by koderholic — APPROVED** [review]
LGTM

**Review by juliaaschmidt — APPROVED** [review]

**Review by mischat — APPROVED** [review]
👍

**Pull request merged**